### PR TITLE
Modify RPC and GEM unpacker blocks in EMTF unpacker to match the new Run 3 format

### PR DESCRIPTION
**Pull request by eyigitba**
#### PR description:

This PR implements the necessary changes to unpack the new DAQ format that is implemented at P5 a couple of weeks ago. Our tests show that the new DAQ format significantly reduces EMTF dead time at very high PU scenarios. According to the new format 2 RPC (GEM) TPs are packed into 1 RPC (GEM) data block, thus reducing the total number of blocks in EMTF DAQ stream.

Modified `EMTFBlockGEM.cc`, `EMTFBlockRPC.cc` to add functionality to create 2 TPs per block depending on the firmware version. Changes in `EMTFSetup.cc` are made to pass firmware version to the RPC and GEM block unpackers.

A backport of this PR to 12_4_X and 12_5_X is required for upcoming collision runs. 

#### PR validation:

Tested with unpacking data collected last week and running `l1tstage2_dqm_sourceclient-live_cfg.py` to confirm the DQM plots are as expected.

### CODE REVIEW

**Comment by cecilecaillol** [comment]
please test

**Comment by rovere** [comment]
Since this is needed for data taking after TS1 and LHC recovery, please open a backport to `12_4` **and** `12_5`. 
On top of that, to make it even more explicit that this is needed for the coming data-taking, please mark the `12_4` backport as `urgent`.

Marco(ORM)

**Comment by cecilecaillol** [comment]
+l1

File: EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockRPC.cc [hunk 1]
```diff
@@ -55,6 +55,15 @@ namespace l1t {
         uint16_t RPCc = payload[2];
         uint16_t RPCd = payload[3];
 
+        // Run 3 has a different EMTF DAQ output format since August 26th
```
Comment (perrotta) [inline EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockRPC.cc:58]: What is the purpose of this commented out block? Please either add a comment which explains its usage, or remove it

Comment (eyigitba) [reply EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockRPC.cc]: This was my bad. I forgot to remove that part :). Should be fixed now

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
@eyigitba I am notincing now that this PR reverts the protections previously added with #38961: is it wanted, or simply you developed in 12_4_X (where that PR was never backported) and copied blindly the same files in 12_5_X and 12_6_X?

I'm holding the backports until this issue is addressed

**Comment by eyigitba** [comment]
@perrotta I can see the protections are still there at https://github.com/cms-sw/cmssw/pull/39388/files#diff-1a6dadb65e6c79d8a2fc124b34a859b8e8536d39e2cdfea6d7d9f05eba4a3bddR262-R269

Are you talking about these?

**Comment by perrotta** [comment]
Ah, ok, thank you for having checked. The confusion was caused by the fact that PR #38961 was never backported in 12_4_X, as its modifications are only there in 12_5_X and 12_6_X

That PR adds the protections in the following files:
 
- EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockGEM.cc (*) 
- EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockME.cc
- EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockRPC.cc (*)
- EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFBlockSP.cc

The backport of this PR adds those protections in EMTFBlockGEM.cc and EMTFBlockRPC.cc, but as such there is no protection for EMTFBlockME.cc and EMTFBlockSP.cc

I think that they could also be needed in 12_4_X, and therefore a partial backport of #38961 with what's missing should be prepared for 12_4_X: @eyigitba what do you think?

**Comment by eyigitba** [comment]
@perrotta I think that's a good point. I can prepare a partial backport of https://github.com/cms-sw/cmssw/pull/38961. 

These protections were only made to protect the DQM code from out-of-range BX values that happened in 2021. The DQM PR is not in 12_4_X, so it's probably not urgent. But it's good to have anyway for consistency

**Pull request merged**